llvm/include/llvm/DebugInfo/CodeView/TypeRecord.h

```diff
@@ -495,7 +495,8 @@ class ClassRecord : public TagRecord {
 };
 
 // LF_UNION
-struct UnionRecord : public TagRecord {
+class UnionRecord : public TagRecord {
+public:
```

Comment:
Why the change here? The use of `struct` _should_ be fine. Or was it that there was a conflict with the declaration from the XMACRO?

Reply:
yes it wasn't usable in the XMACRO to forward declare classes used for the function declarations it creates